runtime/eventlog.c

```diff
@@ -139,12 +139,13 @@ static void setup_eventlog_file()
   eventlog_filename = caml_secure_getenv(T("OCAML_EVENTLOG_PREFIX"));
 
   if (eventlog_filename) {
-    int ret = snprintf_os(output_file, OUTPUT_FILE_LEN, T("%s.%d.eventlog"),
-                         eventlog_filename, Caml_state->eventlog_startup_pid);
+    int ret = snprintf_os(output_file, OUTPUT_FILE_LEN,
+                          T("%s.%" PRIu32 ".eventlog"),
```

Comment:
I'm not sure `PRIu32` is defined by all the versions of MSVC that we care about.  Plus, this is hiding the real problem, which is that a PID is not always a `uint32_t`, on some platforms it is a 64-bit type.

Reply:
Plus, it doesn't even compile (see the CI tests below), probably because `<inttypes.h>` is not included.

Reply:
So what should we use for the return type of `getpid`? Some other parts of the runtime use `int` or `intnat`; is  is `intnat` the right choice?

Reply:
The debugger calls `caml_putword(dbg_out, getpid())`, and `putword` takes an `uint32_t`.

Reply:
The POSIX standards don't specify much about `pid_t` beyond that it's a signed integer type.  In practice, it's generally `int`, except under Solaris and a few other systems where it's `long`.  `long` in 64-bit Solaris is 64-bit wide, so in principle a PID may not be representable as a 32-bit integer, although I doubt this ever happens.  All in all, I would prefer `Caml_state->eventlog_startup_pid` to have type `long` and to be printed as `%ld`.

Reply:
What if we store a `pid_t` in the domain state but cast it to `long` for printing?

Reply:
> What if we store a pid_t in the domain state 

I'd rather not because it would force `<sys/types.h>` to be included in or before `domain_state.h`.  Plus there is no `pid_t` under Windows.